[PICA-103] removed end_block for reward pools

kkast opened the pull request:
https://app.clickup.com/t/867760zbf
removed end_block for reward pools

KaiserKarel commented:
Task linked: [PICA-103 Remove `end_block` pool parameter](https://app.clickup.com/t/20465559/PICA-103)

kkast commented:
@fl-y @easteregg @Dom-Roth  this pr might break some of the fe and tests. I removed end_block field from RewardPool, RewardPoolConfiguration, also removed EndBlockMustBeAfterStartBlock

vimukthi-git reviewed (approved):
LGTM
